src/shared/lending/oracles/Fortress2PoolOracle.sol

```diff
@@ -0,0 +1,206 @@
+// ╚═╝░░░░░╚═╝╚═╝░░╚══╝╚═╝░░╚═╝╚═╝░░╚══╝░╚════╝░╚══════╝
+
+// 
+// 
+// 
+// 
+
+// Github - https://github.com/FortressFinance
+
+import {ERC4626} from "@solmate/mixins/ERC4626.sol";
+import {AggregatorV3Interface} from "@chainlink/src/v0.8/interfaces/AggregatorV3Interface.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {Math} from '@openzeppelin/contracts/utils/math/Math.sol';
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ICurvePool} from "../interfaces/ICurvePool.sol";
+import {IChainlinkAggregator} from "../interfaces/IChainlinkAggregator.sol"; 
+
+
+contract Fortress2PoolOracle is AggregatorV3Interface {
+
+    using SafeCast for uint256;
+
+    ICurvePool public immutable twoPool;
+    ERC4626 public immutable fc2Pool;
+    IChainlinkAggregator public USDC;
+    IChainlinkAggregator public USDT;
+
+    uint256 public lastSharePrice;
+    uint256 public lowerBoundPercentage;
+    uint256 public upperBoundPercentage;
+    uint256 public vaultMaxSpread;
+
+    address public owner;
+
+    bool public isCheckPriceDeviation;
+
+    uint256 constant private DECIMAL_DIFFERENCE = 1e6;
+    uint256 constant private BASE = 1e18;
+
+    /********************************** Constructor **********************************/
+
+    constructor(address _owner) {
+        twoPool = ICurvePool(address(0x7f90122BF0700F9E7e1F688fe926940E8839F353)); 
+        fc2Pool = ERC4626(address(0xe16F15266cD00c418fB63e505361de32ce90Ac9f));
+
+        USDC = IChainlinkAggregator(address(0x50834F3163758fcC1Df9973b6e91f0F0F0434aD3));
+        USDT = IChainlinkAggregator(address(0x3f3f5dF88dC9F13eac63DF89EC16ef6e7E25DdE7));
+
+        lowerBoundPercentage = 20;
+        upperBoundPercentage = 20;
+        
+        owner = _owner;
+
+        uint256 _vaultSpread = fc2Pool.convertToAssets(1e18);
+        vaultMaxSpread = _vaultSpread * 110 / 100; // limit to 10% of the vault spread
+
+        lastSharePrice = uint256(_getPrice());
+
+        isCheckPriceDeviation = true;
+    }
+
+    /********************************** Modifiers **********************************/
+
+    modifier onlyOwner() {
+        if (msg.sender != owner) revert notOwner();
+        _;
+    }
+
+    /********************************** External Functions **********************************/
+
+    function decimals() external pure returns (uint8) {
+        return 18;
+    }
+
+    function description() external pure returns (string memory) {
+        return "fc2Pool USD Oracle";
+    }
+
+    function version() external pure returns (uint256) {
+        return 1;
+    }
+
+    function getRoundData(uint80) external pure returns (uint80, int256, uint256, uint256, uint80) {
+        revert("Not implemented");
+    }
+
+    function latestRoundData() external view returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound) {
+        return (0, _getPrice(), 0, 0, 0);
+    }
```

Comment:
Could we get GMHacker's feedback on the overall design and then implementation? Maybe it would be really beneficial to have another set of experienced eyes on the idea we are building out here. Maybe somebody like him can provide another valuable point of view.

Reply:
We just follow the Chainlink interface here for better composability. Nothing too special. But sure, another pair of eyes is always good.